beacon-chain/core/epoch/precompute/attestation_test.go

```diff
@@ -0,0 +1,242 @@
+		t.Error("did not vote target")
+	}
+	if !votedHead {
+		t.Error("did not vote head")
+	}
+}
+
+func TestAttestedCurrentEpoch(t *testing.T) {
+	deposits, _, _ := testutil.SetupInitialDeposits(t, 100)
+	beaconState, err := state.GenesisBeaconState(deposits, uint64(0), &ethpb.Eth1Data{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	beaconState.Slot = params.BeaconConfig().SlotsPerEpoch + 1
+	att := &ethpb.Attestation{Data: &ethpb.AttestationData{
+		Target:    &ethpb.Checkpoint{Epoch: 1},
+		Crosslink: &ethpb.Crosslink{}}}
+	attSlot, err := helpers.AttestationDataSlot(beaconState, att.Data)
+	if err != nil {
+		t.Fatal(err)
+	}
+	r := []byte{'A'}
+	beaconState.BlockRoots[attSlot] = r
+	att.Data.Target.Root = r
+	att.Data.BeaconBlockRoot = r
+	votedEpoch, votedTarget, err := attestedCurrentEpoch(beaconState, &pb.PendingAttestation{Data: att.Data})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !votedEpoch {
+		t.Error("did not vote epoch")
+	}
+	if !votedTarget {
+		t.Error("did not vote target")
+	}
+}
+
+func TestProcessAttestations(t *testing.T) {
+	helpers.ClearAllCaches()
+
+	params.UseMinimalConfig()
+	validators := uint64(64)
+	deposits, _, _ := testutil.SetupInitialDeposits(t, validators)
+	beaconState, err := state.GenesisBeaconState(deposits, uint64(0), &ethpb.Eth1Data{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	beaconState.Slot = params.BeaconConfig().SlotsPerEpoch
+
+	bf := []byte{0xff}
+	att1 := &ethpb.Attestation{Data: &ethpb.AttestationData{
+		Target:    &ethpb.Checkpoint{Epoch: 0},
+		Crosslink: &ethpb.Crosslink{Shard: 960}}, AggregationBits: bf}
+	att2 := &ethpb.Attestation{Data: &ethpb.AttestationData{
+		Target:    &ethpb.Checkpoint{Epoch: 0},
+		Crosslink: &ethpb.Crosslink{Shard: 961}}, AggregationBits: bf}
+	beaconState.BlockRoots[0] = []byte{'A'}
+	att1.Data.Target.Root = []byte{'A'}
+	att1.Data.BeaconBlockRoot = []byte{'A'}
+	beaconState.BlockRoots[0] = []byte{'B'}
+	att2.Data.Target.Root = []byte{'A'}
+	att2.Data.BeaconBlockRoot = []byte{'B'}
+	beaconState.PreviousEpochAttestations = []*pb.PendingAttestation{{Data: att1.Data, AggregationBits: bf}}
+	beaconState.CurrentEpochAttestations = []*pb.PendingAttestation{{Data: att2.Data, AggregationBits: bf}}
+
+	vp := make([]*Validator, validators)
+	for i := 0; i < len(vp); i++ {
+		vp[i] = &Validator{CurrentEpochEffectiveBalance: 100}
+	}
+	bp := &Balance{}
+	vp, bp, err = ProcessAttestations(context.Background(), beaconState, vp, bp)
+	if err != nil {
+		t.Fatal(err)
+	}
+	indices, _ := helpers.AttestingIndices(beaconState, att1.Data, att1.AggregationBits)
+	for _, i := range indices {
+		if !vp[i].IsPrevEpochAttester {
+			t.Log("Not a prev epoch attester")
+		}
+	}
+	indices, _ = helpers.AttestingIndices(beaconState, att2.Data, att2.AggregationBits)
+	for _, i := range indices {
+		if !vp[i].IsPrevEpochAttester {
+			t.Log("Not a prev epoch attester")
+		}
+		if !vp[i].IsPrevEpochHeadAttester {
+			t.Log("Not a prev epoch head attester")
+		}
+	}
+
+	params.UseMainnetConfig()
```

Comment:
If this is to prevent side effects in other tests, you should do this:
```go
params.UseMinimalConfig()
defer params.UseMainnetConfig()
```

Otherwise a failure in tihs test may cause many failures other tests and slow down the debugging process